flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/InputProcessorUtil.java

```diff
@@ -26,7 +26,14 @@
 import org.apache.flink.runtime.metrics.groups.TaskIOMetricGroup;
 import org.apache.flink.streaming.api.graph.StreamConfig;
 import org.apache.flink.streaming.api.operators.MailboxExecutor;
-import org.apache.flink.streaming.runtime.tasks.SubtaskCheckpointCoordinator;
+import org.apache.flink.streaming.runtime.tasks.checkpointing.AlignedController;
+import org.apache.flink.streaming.runtime.tasks.checkpointing.AlternatingController;
+import org.apache.flink.streaming.runtime.tasks.checkpointing.CheckpointBarrierBehaviourController;
```

Comment:
Looks like `InputProcessorUtil` also belongs to `checkpointing` as it's sole purpose is to create `CheckpointedInputGate`. And the purpose of the latter is to handle events, mostly checkpoint barriers.

How about moving it too along with `UpstreamRecoveryTracker`/Impl and `CheckpointedInputGate`?

WDYT @pnowojski , @AHeise ?

Reply:
"CheckpointedInputGate" is overall an input gate (but just a different type of input gate), that's why I left it in .....runtime.io

`InputProcessorUtil` is mostly createXXXCheckpointedInputGate, that's why I left it with "CheckpointedInputGate"

`UpstreamRecoveryTracker` is channel recovery-related, and that's why I put within .....runtime.io as well.

Reply:
I don't have opinion about this. Could be moved together.

However one thing, I'm not sure if `org.apache.flink.streaming.runtime.tasks.checkpointing` is the best fit. Those classes have very little in common with `task` package and it's content `StreamTask`, `OperatorChain` etc. Why not to move all of it to `org.apache.flink.streaming.runtime.io.checkpointing`?

Reply:
> Those classes have very little in common with task package 

IMO they do have because they run inside (sub)task. In contrast to `CheckpointCoordinator` which runs in JM (and which is in `org.apache.flink.runtime.checkpoint`).

> Why not to move all of it to org.apache.flink.streaming.runtime.io.checkpointing?

I think checkpointing has nothing to with IO except in very general case.

Reply:
Copied my explanation from the Jira ticket why these classes are moved to the package named this way: 

If we do not move SubtaskCheckpointCoordinator(Impl), I think `org.apache.flink.streaming.runtime.checkpointing` is a good choice.

But if we want to move SubtaskCheckpointCoordinator(Impl) together with the above seven classes, I would lean to `org.apache.flink.streaming.runtime.tasks.checkpointing`.

I think it is a good idea to put all checkpointing related classes altogether, so I prefer `org.apache.flink.streaming.runtime.tasks.checkpointing` as a result.